### PR TITLE
feat: add concept of contact card

### PR DESCRIPTION
**Pull request by djaiss**
So we need some sort of contact card. This means when we mouse over a name or an avatar, a small card appears, with some contact information. This will be very useful in a lot of places on Monica.

Here is an example:

To achieve this, we have a new helper, called `ContactCardHelper`, which will put all the information this cards need.

Also, the ContactCard Vue component takes a few parameters. We can choose to display the name of the contact or only the avatar.

Once this PR is accepted, we'll use the component in other places as well - actually, everywhere we display a contact name somewhere in the UI.

### CODE REVIEW

File: domains/Contact/ManageLoans/Web/ViewHelpers/ModuleLoanViewHelper.php [hunk 1]
```diff
@@ -39,17 +40,11 @@ public static function dtoLoan(Loan $loan, Contact $contact, User $user): array
...
         $loaners = $loan->loaners->unique('id');
         $loanees = $loan->loanees->unique('id');
-        $loanersCollection = $loaners->map(function ($loaner) {
-            return [
-                'id' => $loaner->id,
-                'name' => $loaner->name,
-            ];
+        $loanersCollection = $loaners->map(function ($loaner) use ($user) {
+            return ContactCardHelper::data($loaner, $user);
         });
```
Comment (asbiin) [inline domains/Contact/ManageLoans/Web/ViewHelpers/ModuleLoanViewHelper.php:45]: try
```suggestion
        $loanersCollection = $loaners->map(fn ($loaner) => ContactCardHelper::data($loaner, $user));
```

File: domains/Contact/ManageLoans/Web/ViewHelpers/ModuleLoanViewHelper.php [hunk 2]
```diff
@@ -39,17 +40,11 @@ public static function dtoLoan(Loan $loan, Contact $contact, User $user): array
...
-        $loaneesCollection = $loanees->map(function ($loanee) {
-            return [
-                'id' => $loanee->id,
-                'name' => $loanee->name,
-            ];
+        $loaneesCollection = $loanees->map(function ($loanee) use ($user) {
+            return ContactCardHelper::data($loanee, $user);
         });
```
Comment (asbiin) [inline domains/Contact/ManageLoans/Web/ViewHelpers/ModuleLoanViewHelper.php]: same
```suggestion
        $loaneesCollection = $loanees->map(fn ($loanee) => ContactCardHelper::data($loanee, $user));
```

File: tests/Unit/Domains/Contact/ManageLoans/Web/ViewHelpers/ModuleLoanViewHelperTest.php [hunk 3]
```diff
@@ -123,22 +123,12 @@ public function it_gets_the_data_transfer_object(): void
             $array['currency_name']
         );
         $this->assertEquals(
```
Comment (asbiin) [inline tests/Unit/Domains/Contact/ManageLoans/Web/ViewHelpers/ModuleLoanViewHelperTest.php:125]: there is a `$this->assertCount()` method for that

File: resources/js/Shared/ContactCard.vue [hunk 4]
```diff
@@ -0,0 +1,111 @@
+<template>
+  <div class="relative inline" :class="divOuterClass" :style="'top: ' + top">
```
Comment (asbiin) [inline resources/js/Shared/ContactCard.vue]: This `divOuterClass` is useless here. Simply define a `class` attribute on the caller and those classes will be add in the root node (if needed, but you did not use it apparently).

**Review by asbiin — APPROVED** [review]
I love it!
I made some suggestions:

File: resources/js/Shared/ContactCard.vue (hunk 4 above)
Comment (djaiss) [reply resources/js/Shared/ContactCard.vue]: fixed

**Pull request merged**